octodns/manager.py

```diff
@@ -267,16 +268,17 @@ def _populate_and_plan(self, zone_name, sources, targets, desired=None,
         return plans, zone
 
     def sync(self, eligible_zones=[], eligible_sources=[], eligible_targets=[],
-             dry_run=True, force=False):
+             dry_run=True, force=False, plan_output_fh=stdout):
         self.log.info('sync: eligible_zones=%s, eligible_targets=%s, '
-                      'dry_run=%s, force=%s', eligible_zones, eligible_targets,
-                      dry_run, force)
+                      'dry_run=%s, force=%s, plan_output_fh=%s',
+                      eligible_zones, eligible_targets, dry_run, force,
+                      plan_output_fh)
```

Comment:
Looks like the default `__repr__` of fh objects is fairly verbose:

```
>>> with open('/tmp/foo', 'w') as fh:
...     print(f'{fh}')
...
<_io.TextIOWrapper name='/tmp/foo' mode='w' encoding='UTF-8'>
>>> print(f'{stdout}')
<_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
```

Maybe preferable to log `plan_output_fh.name` here? Not 100% sure that file handle objects always have a `.name` though...

```suggestion
                      'dry_run=%s, force=%s, plan_output_fh=%s',
                      eligible_zones, eligible_targets, dry_run, force,
                      plan_output_fh.name)
```

Reply:
Unfortunately, `.name` isn't reliable. One of my primary use-cases is `File`-compatible types like `StringIO`, which does not have that attribute.

Alternatively, I could leave this out of the log line altogether, or indicate whether it's set to the default value at call time. Do either of those options seem better?

Reply:
Maybe something like this:

```suggestion
                      'dry_run=%s, force=%s, plan_output_fh=%s',
                      eligible_zones, eligible_targets, dry_run, force,
                      getattr(plan_output_fh, 'name', plan_output_fh.__class__.__name__))
```

```
>>> fh = StringIO()
>>> getattr(fh, 'name', fh.__class__.__name__)
'StringIO'
>>> with open('/tmp/foo', 'w') as fh:
...     getattr(fh, 'name', fh.__class__.__name__)
...
'/tmp/foo'
```

Otherwise probably make sense to just omit it.